js/cryptocom.js

```diff
@@ -2317,6 +2349,33 @@ module.exports = class cryptocom extends Exchange {
         return rates;
     }
 
+    handleMarginModeAndParams (methodName, params = {}) {
+        /**
+         * @ignore
+         * @method
+         * @description marginMode specified by params["marginMode"], this.options["marginMode"], this.options["defaultMarginMode"], params["margin"] = true or this.options["defaultType"] = 'margin'
+         * @param {object} params extra parameters specific to the exchange api endpoint
+         * @returns {[string|undefined, object]} the marginMode in lowercase
+         */
+        const defaultType = this.safeString (this.options, 'defaultType');
```

Comment:
@Dan-krm I don't think this is the best approach because we're "copying" most of the original `handleMarginModeAndParams` logic here and "extending".  By doing that, we get a lot of duplicated code and if in the future we decide to update `handleMarginModeAndParams` we will need to update it in a lot of places. 

If you want the default behavior plus some custom logic why not something like this?

```Javascript 

handleMarginModeAndParams (methodName, params = {}) {
   
   let marginMode = undefined;
   [marginMode, params] = super.handleMarginModeAndParams(methodName, params); // result from the original implementation

// your custom logic here (check isMargin, etc)
}
```
what do you think?

Reply:
Yeah that makes sense I'll push a PR with some changes